Update README Header and TOC

egillanton opened the pull request:
# IssueType Pull request

- Updated header and TOC in the READEME.md file.

Edited Files:

- README.md

egillanton commented:
I have removed all new files and just requesting pull request for changes for README.md

- Update README.md header
- Update READEME.md TOC

HaukurPall commented:
I agree with Judy:
- The numbered sections look great and give the reader a better overview of the document. Did you manually create these entries? If not, it would be nice to have a reference to the tool.
- Skip the emojis, they are out of character for the rest of the document and I don't believe they fit this type of document.
- Separating "contibuting" to a separate document sounds like something the users should feel free to do, but not what we suggest. We can make our suggestion clearer in the original text.
- Having a "roadmap" in a repo sounds to me like duplication of content, since we use other tools for this now, and will mostly benefit people who stand outside LVL and SÍM, i.e. who do not have access to our tools are meet with us regularly. I believe this group is currently very small. For now, we should skip this section. We might review this decision when we find it useful to add such information to very specific projects.
- LVL should be in the title as the document can be viewed outside this repo.
- The address line should be removed, it provides little extra information to the reader. If they care where LVL is located, they will search for themselves.

SunnevaThorsteins commented:
I agree with both Haukur and Judy.

- The numbered sections are great and will be super helpful to navigate the guidelines.
- The new table of content also looks great. 
- The emojis are also nice but I don't think they fit with this kind of document and therefore should be removed. 
- LVL should be in the title to make it clear whom these guidelines are for and help avoid confusion. Even though it's in the line below it's easy to miss. 
- I don't think including a contribution and a roadmap file belongs as a base suggestion since it varies from project to project weather they are useful or not. Including them or not should be up to the developers. 
-  The address line isn't necessary since this document is mostly meant for people within LVL and others interested can easily find out elsewhere.

judyfong commented:
@egillanton Do you have time to make the changes by the end of week? Then we can merge and close this PR.

egillanton commented:
I have now: 

- Renamed the Title of the README to `LVL Software Development Guidelines` 
- Updated the TOC using [markdown-toc](http://ecotrust-canada.github.io/markdown-toc)
- Removed the Emoji in the Title
- Removed Line 8: "🚧 Under construction 🚧"

judyfong commented on readme.md:
```diff
@@ -1,168 +1,112 @@
+<p align="center"><i>
+  This repository contains the software development guidelines for LVL.<br/>
+  Center for Analysis and Design of Intelligent Agents, Language and Voice Lab <br/>
+  Reykjavik University - School of Computer Science, Menntavegur 1, IS-101 Reykjavik, Iceland
```
```suggestion
  [Reykjavik University](https://ru.is)
```

judyfong replied:
This might be a way for people to easily find out the university's address while not including it in this document at the same time.

egillanton replied:
Done

judyfong commented on readme.md:
```diff
@@ -384,17 +479,48 @@ This can be reduced somewhat by targetting very specific tasks early in the deve
 ### 8.1. What is Continuous integration?
-Continuous integration is a way to **build software, automaticly run tests and alert developers when a test doesn't pass**. This way developers can fix mistakes before they pile up and become difficult to manage. It also gives developers more confidance in their code and minimizes unknown errors. 
+Continuous integration is a way to **build software (compile), automatically run [tests](#testing), [lint and check documentation](#linting) and alert developers when something is wrong**.
+We suggests running all these steps automatically in the CI system.
```
```suggestion
We suggest running all these steps automatically in the CI system.
```

egillanton replied:
Done

judyfong commented on readme.md:
```diff
@@ -246,7 +211,7 @@ Main authors <email.addresses>
 If the funding is from a public grant, mention the source of the funding and link to their website.
 
-"This project was funded by the Language Technology Programme for Icelandic 2019-2023. The programme, which is managed and coordinated by [Almannarómur](https://almannaromur.is/), is funded by the Icelandic Ministry of Education, Science and Culture."
+214: "This project was funded by the Language Technology Programme for Icelandic 2019-2023. The programme, which is managed and coordinated by [Almannarómur](https://almannaromur.is/), is funded by the Icelandic Ministry of Science and Culture."
```
```suggestion
"This project was funded by the Language Technology Programme for Icelandic 2019-2023. The programme, which is managed and coordinated by [Almannarómur](https://almannaromur.is/), is funded by the Icelandic Ministry of Education, Science and Culture."
```

judyfong replied:
A bunch of lines also contain the line number for whatever reason. I've pointed some out but you should also fix the others.

egillanton replied:
Done

judyfong replied:
It's still missing Education,

HaukurPall replied:
When I updated this PR with the latest changes from master this is in a separate file.

judyfong commented on readme.md:
```diff
@@ -277,7 +242,7 @@ We want our code to be freely available for everyone so we prefer permissive lic
 - CC BY 4.0 (for resources)
 
-For more help choosing a license, see [Choosing a license](https://choosealicense.com/). Keep in mind that if you are working through SÍM you have agreed to use open licenses such as these. Furthermore, according to RU contracts you are the copyright owner.
+245: For more help choosing a license, see [Choosing a license](https://choosealicense.com/). Keep in mind that if you are working through SÍM you have agreed to use open licenses such as these. Furthermore, according to RU are the copyright owner.
```
```suggestion
For more help choosing a license, see [Choosing a license](https://choosealicense.com/). Keep in mind that if you are working through SÍM you have agreed to use open licenses such as these. Furthermore, according to RU contracts you are the copyright owner.
```

egillanton replied:
Done

judyfong replied:
still missing contracts you

HaukurPall replied:
When I updated this PR with the latest changes from master this is there.

judyfong commented on readme.md:
```diff
@@ -1,160 +1,129 @@
+<p align="center"><i>
+This repository contains the software development guidelines for LVL.<br/>
+Center for Analysis and Design of Intelligent Agents, Language and Voice Lab <br/>
+Reykjavik University - School of Computer Science, Menntavegur 1, IS-101 Reykjavik, Iceland
```
```suggestion
[Reykjavik University](https://ru.is)
```

egillanton replied:
Done

egillanton commented:
Should be good to go.

judyfong commented on readme.md:
```diff
@@ -286,38 +246,116 @@ When a license has been chosen, simply copy the license text, fill in any additi
 Writing documentation can be hard but we have some suggestions to make it easier, but keep in mind that a good `#Running` section the `README.md` is complementary to the documentation, i.e. you need to write both.
 - The `#Running` section handles common use-cases. This can be seen as a small [user guide](#user-guide).
-- Write the documentation for functions and classes in your code using your language's conventions since all (common) languages have tools which can extract this documentation. This documentation is more detailed than the `#Running` section but can also contain the same examples.
+- Write the documentation for functions and classes in your code using your language's conventions since all (common) languages have tools which can extract this documentation. This documentation is more detailed than the  section but can also contain the same examples.
```
```suggestion
- Write the documentation for functions and classes in your code using your language's conventions since all (common) languages have tools which can extract this documentation. This documentation is more detailed than the `#Running` section but can also contain the same examples.
```

HaukurPall replied:
When I updated this PR with the latest changes from master I noticed that this issue is no longer there.

judyfong commented on readme.md:
```diff
@@ -326,25 +364,21 @@ For example:
+### 7.1. Workflow
+If a project is public, **the master branch should always contain production ready code**. All merges to the master branch should include semantic version tags as listed below.
 
-For larger projects, [Gitflow](https://www.atlassian.com/git/tutorials/comparing-workflows/gitflow-workflow) is the recommended workflow. It includes **two main branches**, the master branch that stores release history and a development branch that lives between the master branch and feature branches. It also recommends a clear way to implement hotfixes.
```
Re-add the comment about larger projects.

judyfong commented on readme.md:
```diff
@@ -359,35 +393,34 @@ To clearly state the benefits of this, if enforced:
 - Minor: Added backwards compatible functionality, ***when you add features***
 - Patch: Backwards compatible fixes, ***when you fix bugs***
 
-[Read more about semantic versioning](https://semver.org/) and [how to use tags to apply semantic versioning](https://www.atlassian.com/git/tutorials/inspecting-a-repository/git-tag)
```
Don't delete this line

HaukurPall replied:
When I updated this PR with the latest changes from master this is there.

judyfong commented on readme.md:
```diff
@@ -418,141 +451,141 @@ This part could be done automatically but is often time consuming and difficlut
-## Style
-Maintaining the same style across a project makes the code more readable. We suggest using a tool which automatically formats/styles your code. This eliminates inconsistent styles in the code and allows you to focus on the code, rather than the format.
-
-### Python
-We highly suggest [`black`](https://github.com/psf/black)
-
```
Don't delete these lines

HaukurPall replied:
When I updated this PR with the latest changes from master I moved this to a better place.

judyfong commented on readme.md:
```diff
@@ -418,141 +451,141 @@ This part could be done automatically but is often time consuming and difficlut
+### 11.1. What is Continuous integration?
 Continuous integration is a way to **build software (compile), automatically run [tests](#testing), [lint and check documentation](#linting) and alert developers when something is wrong**.
-We suggests running all these steps automatically in the CI system.
-This way developers can fix mistakes before they pile up and become difficult to manage.
-It also gives developers more confidance in their code and minimizes unknown errors. 
```
Don't delete these lines.

HaukurPall replied:
When I updated this PR with the latest changes from master this is there.

judyfong commented on readme.md:
```diff
@@ -418,141 +451,141 @@ This part could be done automatically but is often time consuming and difficlut
+#### 11.2.1. Triggering workflows
+Your workflow should run at minimum each time a push or a pull request is made to the master. It is also recomended to run a [scheduled](https://help.github.com/en/actions/reference/trigger-workflows#scheduled-events-schedule) workflow once a day for building, testing and packaging projects. If workflows take a long time to run it is advisable to run them over night. 
+
+#### 11.2.2. Jobs
+Jobs should at minimum include [testing](#testing) and [packaging](#packaging--releasing) for the project.
+
+Multiple jobs can be specified within each workflow. Besides testing and packaging, which are required, it is also recommended to run a linter on the project to ensure consistent styling across the project.
+
+Keep in mind that each job may take no longer than 6 hours to complete or it will be automatically ended.
+
+When running a job we need to specify which machine it [runs on](https://help.github.com/en/actions/reference/workflow-syntax-for-github-actions#jobsjob_idruns-on). It is **strongly recommended to use github hosted 
```
This has been deleted in the revision

HaukurPall replied:
When I updated this PR with the latest changes from master this is not there.

judyfong commented on readme.md:
```diff
@@ -418,141 +451,141 @@ This part could be done automatically but is often time consuming and difficlut
-  [push, pull_request]:
-    # Which branch/es the action will run on
-    branches: [ master ]
+# The event/s you want to trigger this action
+[push, pull_request]:
+# Which branch/es the action will run on
+branches: [ master ]
```
It'd be good to maintain the indentation

HaukurPall replied:
When I updated this PR with the latest changes from master this has been moved to a separate file.

HaukurPall commented on readme.md:
```diff
@@ -277,7 +237,7 @@ We want our code to be freely available for everyone so we prefer permissive lic
 - CC BY 4.0 (for resources)
 
-For more help choosing a license, see [Choosing a license](https://choosealicense.com/). Keep in mind that if you are working through SÍM you have agreed to use open licenses such as these. Furthermore, according to RU contracts you are the copyright owner.
+For more help choosing a license, see [Choosing a license](https://choosealicense.com/). Keep in mind that if you are working through SÍM you have agreed to use open licenses such as these. Furthermore, according to RU are the copyright owner.
```
```suggestion
For more help choosing a license, see [Choosing a license](https://choosealicense.com/). Keep in mind that if you are working through SÍM you have agreed to use open licenses such as these. Furthermore, according to RU, you are the copyright owner.
```

HaukurPall commented on readme.md:
```diff
@@ -286,38 +246,116 @@ When a license has been chosen, simply copy the license text, fill in any additi
+#### 5.4.2. Tools
+[List of python tools to auto generate documentation](https://wiki.python.org/moin/DocumentationTools)
+
+## 6. Documentation
```
This looks like duplicated sections (5 and 6). I can fix this in another PR, so I'm ok with this for now.

HaukurPall commented on readme.md:
```diff
@@ -286,38 +246,116 @@ When a license has been chosen, simply copy the license text, fill in any additi
 
 ---
 
-## Documentation
```
This is the old documentation section

HaukurPall commented on readme.md:
```diff
@@ -286,38 +246,116 @@ When a license has been chosen, simply copy the license text, fill in any additi
 - Tool: [`Sphinx` with `napoleon`](https://www.sphinx-doc.org/en/master/usage/extensions/napoleon.html)
 
-#### JavaScript
+#### 6.4.2. JavaScript
```
This should probably be 6.1.2

HaukurPall commented on readme.md:
```diff
@@ -286,38 +246,116 @@ When a license has been chosen, simply copy the license text, fill in any additi
+- This could be implemented as a long `#Running` section.
+- If working with the command-line your program should support `--help / -h` commands which should offer some text explaining what the script does, examples and description of parameters. For Python [argparse](https://docs.3/library/argparse.html) and [click](https://click.palletsprojects.com/en/7.x/) make it easy to get this functionality.
+
+#### 6.4.1. Python
```
This should probably be 6.1.1

HaukurPall commented on readme.md:
```diff
@@ -286,38 +246,116 @@ When a license has been chosen, simply copy the license text, fill in any additi
 Place this information into `Contributing.md`.
 
-### User guide
+### 6.7. User guide
```
I am confused, how does the HEAD actually look in this branch? Isn't 6.4 about the user guide? This comment also applies to 6.6 and 6.5.

HaukurPall commented on readme.md:
```diff
@@ -326,25 +364,21 @@ For example:
----
-
-## Version control
-Within Cadia-LVL, using git for version control is required. A clean repository with descriptive commit messages makes for a good representation of your project which makes it easier for new developers to join it. The following guidelines describe a way to maintain a project as such.
```
Old text should be updated.

HaukurPall commented on readme.md:
```diff
@@ -326,25 +364,21 @@ For example:
 - This could be a video guide which go through the same steps as above.
 - This could be implemented as a Wiki on GitHub.
 - This could be implemented as a long `#Running` section.
-- If working with the command-line your program should support `--help / -h` commands which should offer some text explaining what the script does, examples and description of parameters. For Python [argparse](https://docs.python.org/3/library/argparse.html) and [click](https://click.palletsprojects.com/en/7.x/) make it easy to get this functionality.
```
Old text should be updated

HaukurPall commented on readme.md:
```diff
@@ -418,141 +451,141 @@ This part could be done automatically but is often time consuming and difficlut
 User testing is a great way to see how the users interact with your product. 
 However, user testing is difficult to do well. When doing user tests keep the following points in mind.
 
-#### User tests should only be used if you have planned time to make adjustments
```
The following lines look strange...
